File: OCP/Java/BadOCP/InterestCalculator.java
public class InterestCalculator {

    public double calculate(String accountType, double balance) {

        if (accountType.equals("SAVINGS")) {
            return balance * 0.04;
        } else if (accountType.equals("CURRENT")) {
            return 0;
        } else if (accountType.equals("LOAN")) {
            return balance * 0.08;
        }

        return 0;
    }

    public static void main(String[] args) {
        InterestCalculator calc = new InterestCalculator();
        System.out.println(calc.calculate("SAVINGS", 10000));
    }
}

File: OCP/TS/BadOCP/InterestCalculator.js
var InterestCalculator = /** @class */ (function () {
    function InterestCalculator() {
    }
    InterestCalculator.prototype.calculate = function (type, balance) {
        if (type === "SAVINGS")
            return balance * 0.04;
        if (type === "CURRENT")
            return 0;
        if (type === "LOAN")
            return balance * 0.08;
        return 0;
    };
    return InterestCalculator;
}());
var calc = new InterestCalculator();
console.log(calc.calculate("SAVINGS", 10000));
